ouroboros-network/src/Ouroboros/Network/PeerSelection/Governor/Monitor.hs

```diff
@@ -160,10 +170,17 @@ connections PeerSelectionActions{
                          PeerWarm -> ( peerState
                                      , (\x -> (x + realToFrac aFuzz) `max` 0) <$> reconnectDelay
                                      )
-                         PeerCold -> ( peerState
+                         PeerCooling -> ( peerState
                                      , (\x -> (x + realToFrac rFuzz) `max` 0) <$> reconnectDelay
                                      )
+                         PeerCold -> a
                        ) <$> demotions
+      -- update cache by removing the ones that were demoted to really cold
+      -- and adding the ones demoted to cold.
+      modifyTVar coolingPeersCacheTVar
+                 ( (Set.\\ Map.keysSet demotedToCold)
+                 . (<> Map.keysSet demotedToCoolingOrCold)
+                 )
       return $ \now ->
```

Comment:
With a cache we have to be very careful when things enter it and when things exit it so we don't introduce a memory leak.  The code is fine because eventually every connection becomes `Cold` so it will leave the cache, so it's fine.

The comment above is using _really cold_ which should just be _cold_, isn't it?